src/NaCl.Core/Base/ChaChaCore/ChaCha20CoreIntrinsics.cs

```diff
@@ -0,0 +1,54 @@
+﻿#if INTRINSICS
```

Comment:
I wonder if we do need the preprocessor directive since we are checking if the hardware supports it